### ISSUE TITLE
Consider making SuperOffice.WebApi.IdentityModel.SystemUserClient async

### ISSUE THREAD

**Issue by stonesvillage** [SuperOffice/SuperOffice.WebApi-Samples#10]
Hi,
From what I can see the `SuperOffice.WebApi.IdentityModel.SystemUserClient` isn't async and can cause deadlocks.

Regards

Gunnar Stensby

**Comment by stonesvillage** [comment]
Hi,

This package is great and we're using it in all of our apps that is based on .net core 3.1.

Started testing it yesterday in apps running net472 and now I get these deadlocks I mentioned in this issue a couple of weeks ago.
The main reason for us to use the package in net472 is to execute agents in a systemuser context.

Example in a ASP.NET WebApi (.NET Framework 4.7.2) project where it fails:
```c#
  public class HomeController : Controller
  {
      public async Task<ActionResult> Index()
      {
          var loggerFactory = new NullLoggerFactory();
          var logger = loggerFactory.CreateLogger<HomeController>();
          var systemUserInfo = CreateSystemUserInfo();
          var systemUserClient = new SystemUserClient(systemUserInfo, new HttpClient(), logger);
          var ticket = await System.Threading.Tasks.Task.Run(() => systemUserClient.GetSystemUserTicket()); // no deadlock

          //ticket = systemUserClient.GetSystemUserTicket(); // deadlock!

          return Content(ticket);
      }
}
```
As you can see running the code on the ThreadPool instead fixes the deadlock problem (but can come with other problems...).

The code causing the deadlock is probably in class `SuperOffice.WebApi.IdentityModel.SystemUserClient`, method `GetSystemUserJwt()`: 
```c#
HttpResponseMessage result = client.SendAsync(request).Result;
```
Making the code async should solve the deadlock problem: 
```c#
HttpResponseMessage result = await client.SendAsync(request).ConfigureAwait(false);
```

I think there is more examples of blocking code in the code base so going through might be a good idea.

Any chance that this will be fixed any time soon?

Regard

Gunnar Stensby, i-Centrum AB

**Comment by SuperOfficeDevNet** [comment]
Hi Gunnar,

I sympathize with your position. I do. I have fixed the issue in a standalone library, but not commited it to the main internal repo. 

When using the library you would ideally like to ask for a ticket like this:

var ticket = await systemUserClient.GetSystemUserTicket();

But it is not async, and which is why the method is **not** called GetSystemUserTicket**Async**().

In the latest code (not-yet-published), it is async does have the Async suffix, and async/await users will be able to write it as:

var ticket = await systemUserClient.GetSystemUserTicketAsync();

Non-async users will have to write:

var ticket = await systemUserClient.GetSystemUserTicketAync().GetAwaiter().GetResult();

As soon as I can...

**Comment by stonesvillage** [comment]
Hi,

Sounds great with a standalone package, is there a roadmap available?

I understand that the method isn't async but it using async api:s in a sync way under the hoods -> deadlock in code running in the context of a SynchronizationContext (legacy ASP.NET MVC5 for example).

Please go through the existing SuperOffice.WebApi code base as well to eliminate deadlock issues, I've seen some examples in decompiled code.

Thanks!

**Comment by SuperOfficeDevNet** [comment]
Ok. So we finally released SuperOffice.WebApi! The newly published version is 9.2.7935, falls in line with existing SuperOffice versioning, minus the build number.

Starting with version 10, all packages will start to adopt similar versioning, to be more inline with SemVer standards.

Closing this issue as the Async methods are now released.

**Comment by stonesvillage** [comment]
Great - thanks!
